File: files/2025-07-27/CSEC 27072025.py
import pandas as pd

path = "files/2025-07-27/CSEC 27072025.xlsx"
input = pd.read_excel(path, usecols="B:N", skiprows=1, nrows=3)
requirements = input.iloc[0].to_frame('Requirements')
test = input.iloc[1].to_frame('Test')
requirements['New Purchase'] = (req := requirements['Requirements']).sub(req.cummax().shift(fill_value=0)).clip(lower=0)

print(all(test['Test'] == requirements['New Purchase'])) # True

File: files/2025-07-27/CSEC 27072025.R
library(tidyverse)
library(readxl)

path = "files/2025-07-27/CSEC 27072025.xlsx"
input = read_excel(path, range = "B3:N3", col_names = FALSE) %>% t() %>% data.frame(Requirements = .)
test  = read_excel(path, range = "B4:N4", col_names = FALSE) %>% t() %>% data.frame(Test = .)

result = input %>%
  mutate(`New Purchase` = pmax(Requirements - lag(cummax(Requirements),1, default = 0),0))

all.equal(result$`New Purchase`, test$Test, check.attributes = FALSE) 
# > [1] TRUE